Exiled.Events/Patches/Events/Player/ChangingMicroHIDState.cs

```diff
@@ -0,0 +1,42 @@
+// -----------------------------------------------------------------------
+// <copyright file="ChangingMicroHIDState.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Player
+{
+#pragma warning disable SA1313
+
+    using Exiled.API.Features;
+    using Exiled.Events.EventArgs;
+
+    using HarmonyLib;
+
+    /// <summary>
+    /// Patches <see cref="MicroHID.NetworkCurrentHidState"/>.
+    /// Adds the <see cref="Handlers.Player.OnChangingMicroHIDState"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(MicroHID), nameof(MicroHID.NetworkCurrentHidState), MethodType.Setter)]
+    internal static class ChangingMicroHIDState
+    {
+        private static bool Prefix(MicroHID __instance, ref MicroHID.MicroHidState value)
+        {
+            // NetworkCurrentHid state is set each frame, so this is to prevent calling the method each frame.
+            if (__instance.CurrentHidState == value)
+                return true;
```

Comment:
In .NET property setters not ever called when assigning the same value 👀

Reply:
~~Northwood moment~~
Without this, the method is called each frame so maybe it isn't a property fault as you said, but something else.
Anyways without this `if` the event doesn't work as it should.

Reply:
👌 then keep it up

Reply:
> In .NET property setters not ever called when assigning the same value 👀
Aren't they? I tested it pretty quickly and they seem to be called even when assigning to them the same value.

Reply:
> Aren't they? To be 100% sure I tested it and they get called even when assigning to them the same value.

Dunno, that was an old thread